src/rp2040/hardware_structs/include/hardware/structs/clocks.h

```diff
@@ -32,41 +36,288 @@ enum clock_index {
+    // 0x000000ff [0-7]   : FRAC (0): Fractional component of the divisor
     io_rw_32 div;
-    io_rw_32 selected;
+
+    _REG_(CLOCKS_CLK_GPOUT0_SELECTED_OFFSET)
+    // Indicates which SRC is currently selected by the glitchless mux (one-hot)
+    io_ro_32 selected;
+
 } clock_hw_t;
 /// \end::clock_hw[]
 
 typedef struct {
+    _REG_(CLOCKS_CLK_SYS_RESUS_CTRL_OFFSET)
+    // 0x00010000 [16]    : CLEAR (0): For clearing the resus after the fault that triggered it has...
+    // 0x00001000 [12]    : FRCE (0): Force a resus, for test purposes only
+    // 0x00000100 [8]     : ENABLE (0): Enable resus
+    // 0x000000ff [0-7]   : TIMEOUT (0xff): This is expressed as a number of clk_ref cycles
+    io_rw_32 ctrl;
+
+    _REG_(CLOCKS_CLK_SYS_RESUS_STATUS_OFFSET)
+    // 0x00000001 [0]     : RESUSSED (0): Clock has been resuscitated, correct the error then send ctrl_clear=1
+    io_ro_32 status;
+
+} clock_resus_hw_t;
+
+typedef struct {
+    _REG_(CLOCKS_FC0_REF_KHZ_OFFSET)
+    // Reference clock frequency in kHz
+    // 0x000fffff [0-19]  : FC0_REF_KHZ (0)
     io_rw_32 ref_khz;
+
+    _REG_(CLOCKS_FC0_MIN_KHZ_OFFSET)
+    // Minimum pass frequency in kHz
+    // 0x01ffffff [0-24]  : FC0_MIN_KHZ (0)
     io_rw_32 min_khz;
+
+    _REG_(CLOCKS_FC0_MAX_KHZ_OFFSET)
+    // Maximum pass frequency in kHz
+    // 0x01ffffff [0-24]  : FC0_MAX_KHZ (0x1ffffff)
     io_rw_32 max_khz;
+
+    _REG_(CLOCKS_FC0_DELAY_OFFSET)
+    // Delays the start of frequency counting to allow the mux to settle
+    // 0x00000007 [0-2]   : FC0_DELAY (1)
     io_rw_32 delay;
+
+    _REG_(CLOCKS_FC0_INTERVAL_OFFSET)
+    // The test interval is 0
+    // 0x0000000f [0-3]   : FC0_INTERVAL (0x8)
     io_rw_32 interval;
+
+    _REG_(CLOCKS_FC0_SRC_OFFSET)
+    // Clock sent to frequency counter, set to 0 when not required
+    // 0x000000ff [0-7]   : FC0_SRC (0)
     io_rw_32 src;
+
+    _REG_(CLOCKS_FC0_STATUS_OFFSET)
+    // Frequency counter status
+    // 0x10000000 [28]    : DIED (0): Test clock stopped during test
+    // 0x01000000 [24]    : FAST (0): Test clock faster than expected, only valid when status_done=1
+    // 0x00100000 [20]    : SLOW (0): Test clock slower than expected, only valid when status_done=1
+    // 0x00010000 [16]    : FAIL (0): Test failed
+    // 0x00001000 [12]    : WAITING (0): Waiting for test clock to start
+    // 0x00000100 [8]     : RUNNING (0): Test running
+    // 0x00000010 [4]     : DONE (0): Test complete
+    // 0x00000001 [0]     : PASS (0): Test passed
     io_ro_32 status;
+
+    _REG_(CLOCKS_FC0_RESULT_OFFSET)
+    // Result of frequency measurement, only valid when status_done=1
+    // 0x3fffffe0 [5-29]  : KHZ (0)
+    // 0x0000001f [0-4]   : FRAC (0)
     io_ro_32 result;
+
 } fc_hw_t;
 
 typedef struct {
-    clock_hw_t clk[CLK_COUNT];
-    struct {
-        io_rw_32 ctrl;
-        io_rw_32 status;
-    } resus;
+    clock_hw_t clk[10];
```

Comment:
I know that this new code is auto-generated (and so it will always have the "correct value"), but I wonder if having `clock_hw_t clk[CLK_COUNT];` instead of `clock_hw_t clk[10];` is more "semantically useful" to the reader? :thinking: 
Same comment also applies to e.g. `dma_channel_hw_t ch[NUM_DMA_CHANNELS];` versus `dma_channel_hw_t ch[12];`.
(Feel free to ignore this suggestion if it would be too difficult to change!)

Reply:
I discussed this with Luke, and they felt that it was fine to keep it as a number. I could add it without much issue as it would just be another param to be pulled in from the mapping file.